unity/acsShowcase/Assets/Scripts/Calling/CallScenario.cs

```diff
@@ -213,14 +213,28 @@ public CallAgent CallAgent
             {
                 if (currentCallAgent != null)
                 {
-                    currentCallAgent.IncomingCallReceived -= OnCurrentCallAgentIncomingCall;
+                    if (currentCallAgent is TeamsCallAgent teamsCallAgent)
```

Comment:
Sort of a nit....To help readability can you split the event registration adding and removing into separate functions:

```cs
        /// <summary>
        /// Add event handlers to the call agent.
        /// </summary>
        private void AddCallAgentEventHandlers(CommonCallAgent commonCallAgent)
        {
            if (commonCallAgent != null)
            {
                if (commonCallAgent is CallAgent callAgent)
                {
                    callAgent.IncomingCallReceived += OnIncomingCall;
                }
                else if (commonCallAgent is TeamsCallAgent teamsCallAgent)
                {
                    teamsCallAgent.IncomingCallReceived += OnIncomingTeamsCall;
                }
            }
        }
        /// <summary>
        /// Remove event handlers from the call agent.
        /// </summary>
        private void RemoveCallAgentEventHandlers(CommonCallAgent commonCallAgent)
        {
            if (commonCallAgent != null)
            {
                if (commonCallAgent is CallAgent callAgent)
                {
                    callAgent.IncomingCallReceived -= OnIncomingCall;
                }
                else if (commonCallAgent is TeamsCallAgent teamsCallAgent)
                {
                    teamsCallAgent.IncomingCallReceived -= OnIncomingTeamsCall;
                }
            }
        }
```